app/controllers/Application.scala

```diff
@@ -55,91 +82,167 @@ final class Application @Inject()(data: DataHelper,
+        }
+      }
 
-      if (categoryArray != null && Categories.visible.toSet.equals(categoryArray.toSet))
-        categoryArray = null
+      list.map { case (v, p, c, a, u) =>
+        (v, p, c, a, u, reviewData.getOrElse(v.id.get, None))
+      }
+    }
+    data map { list =>
+      val lists = list.partition(_._6.isEmpty)
+      val reviewList = lists._2.map { case (v, p, c, a, _, r) =>
+        (p, v, c, a, r.get)
+      }
+      val unReviewList = lists._1.map { case (v, p, c, a, u, _) =>
+        (p, v, c, a, u)
+      }
 
-      Ok(views.home(projects, Option(categoryArray), query.find(_.nonEmpty), p, ordering, pform))
+      Ok(views.users.admin.queue(reviewList, unReviewList))
     }
+
   }
 
-  /**
-    * Show external link warning page.
-    *
-    * @return External link page
-    */
-  def linkOut(remoteUrl: String) = {
-    Action { implicit request =>
-      Ok(views.linkout(remoteUrl))
+  private def queryReviews(versions: Seq[Int]) = {
+
+    val reviewsTable = TableQuery[ReviewTable]
+    val userTable = TableQuery[UserTable]
+    for {
+      r <- reviewsTable if r.versionId inSetBind versions
+      u <- userTable if r.userId === u.id
+    } yield {
+      (r, u.name)
     }
+
   }
 
-  /**
-    * Shows the moderation queue for unreviewed versions.
-    *
-    * @return View of unreviewed versions.
-    */
-  def showQueue() = {
-    (Authenticated andThen PermissionAction[AuthRequest](ReviewProjects)) { implicit request =>
-      Ok(views.users.admin.queue(
-        this.service.access[Version](classOf[Version])
-          .filterNot(_.isReviewed)
-          .filterNot(_.channel.isNonReviewed)
-          .filterNot(_.reviewEntries.isEmpty)
-          .map(v => (v.project, v)),
-        this.service.access[Version](classOf[Version])
-          .filterNot(_.isReviewed)
-          .filterNot(_.channel.isNonReviewed)
-          .filter(_.reviewEntries.isEmpty)
-          .map(v => (v.project, v))))
+  private def queryQueue = {
+    val versionTable = TableQuery[VersionTable]
+    val channelTable = TableQuery[ChannelTable]
+    val projectTable = TableQuery[ProjectTableMain]
+    val userTable = TableQuery[UserTable]
+
+    for {
+      (v, u) <- versionTable joinLeft userTable on (_.authorId === _.id)
+      c <- channelTable if v.channelId === c.id && v.isReviewed =!= true
+      p <- projectTable if v.projectId === p.id
+      ou <- userTable if p.userId === ou.id
+    } yield {
+      (v, p, c, u.map(_.name), ou)
     }
+
   }
 
   /**
     * Shows the overview page for flags.
     *
     * @return Flag overview
     */
-  def showFlags() = FlagAction { implicit request =>
-    val flags = this.service.access[Flag](classOf[Flag]).filterNot(_.isResolved)
-    Ok(views.users.admin.flags(flags))
+  def showFlags() = FlagAction.async { implicit request =>
+    for {
+      flags <- this.service.access[Flag](classOf[Flag]).filterNot(_.isResolved)
+      users <- Future.sequence(flags.map(_.user))
```

Comment:
before that make it less queries. 1:1 relationship flag-user-project can be queried in a single query instead of 1+2n in this case.
same for a lot of cases where I used Future.sequence